### DISCUSSION TITLE
Command palette for discoverability

### DISCUSSION THREAD

**Discussion by samholmes** [jesseduffield/lazygit#3134] [Ideas]
I propose a command palette activated by `?`. Right now `?` brings up a command list which is a helpful way to lookup keybindings. I recommend that it not only shows you the list of keybindings, but allows for you to fuzzy search for a command to run from this modal. This acts a bit like Cmd+p on other systems like VSCode. This modal is all you'd need to discover all the available commands under any specific context (state) that you might be in. This list of commands have a description which, as mentioend, can be fuzzy searched. The command internally would have an identifier as well (which may also be shown to the right of the description and may also be fuzzy searched). This internal command identifier can be mapped to a specific keybinding from a config file. The keybinding of course would be scoped to a certain state (mode). With this, all the available features of lazygit are discoverable and learnable from the help/command palette and the only thing a user would need to learn to start using lazygit is to press `?`.

**Comment by ekil1100** [comment]
Right now, you can using `/` to search under `?` mode, then `enter` to excute it. But I agree a command center will make more sense. +1
